modules/gnome-extensions/gnome-extensions.sh

```diff
@@ -3,88 +3,226 @@
+      URL_QUERY=$(curl -s "https://extensions.gnome.org/extension-query/?search=${WHITESPACE_HTML}")
+      QUERIED_EXT=$(echo "${URL_QUERY}" | yq ".extensions[] | select(.name == \"${UNINSTALL_EXT}\")")
+      if [[ -z "${QUERIED_EXT}" ]]; then
+        echo "ERROR: Extension '${UNINSTALL_EXT}' does not exist in https://extensions.gnome.org/ website"
+        echo "       Extension name is case-sensitive, so be sure that you typed it correctly,"
+        echo "       including the correct uppercase & lowercase characters"
+        exit 1
+      fi
+      EXT_UUID=$(echo "${QUERIED_EXT}" | yq ".uuid")
+      EXT_NAME=$(echo "${QUERIED_EXT}" | yq ".name")
+      # This is where uninstall step goes, above step is reused from install part
+      EXT_FILES="/usr/share/gnome-shell/extensions/${EXT_UUID}"
+      UNINSTALL_METADATA="${EXT_FILES}/metadata.json"
+      GETTEXT_DOMAIN=$(yq -I=0 ".gettext-domain" < "${UNINSTALL_METADATA}")
+      SETTINGS_SCHEMA=$(yq -I=0 ".settings-schema" < "${UNINSTALL_METADATA}")
+      LANGUAGE_LOCATION="/usr/share/locale"
+      # If settings-schema YAML key exists, than use that, if it doesn't
+      # Than substract the schema ID before @ symbol
+      if [[ ! "${SETTINGS_SCHEMA}" == "null" ]]; then
+        SCHEMA_LOCATION="/usr/share/glib-2.0/schemas/${SETTINGS_SCHEMA}.gschema.xml"
+      else
+        SUBSTRACTED_UUID=$(echo "${EXT_UUID}" | cut -d'@' -f1)
+        SCHEMA_LOCATION="/usr/share/glib-2.0/schemas/org.gnome.shell.extensions.${SUBSTRACTED_UUID}.gschema.xml"
+      fi  
+      # Remove languages
+      if [[ ! "${GETTEXT_DOMAIN}" == "null" ]]; then
+        find "${LANGUAGE_LOCATION}" -type f -name "${GETTEXT_DOMAIN}.mo" -exec rm {} \;
+      else
+        echo "There are no extension languages to remove, since extension doesn't contain them"
+      fi
+      # Remove gschema xml
+      if [[ ! "${SETTINGS_SCHEMA}" == "null" ]] && [[ -f "${SCHEMA_LOCATION}" ]]; then
+        rm "${SCHEMA_LOCATION}"
+      else
+        echo "There is no gschema xml to remove, since extension doesn't have any settings"
+      fi
+      # Removing main extension files
+      if [[ -d "${EXT_FILES}" ]]; then
+        echo "Removing main extension files"
+        rm -r "${EXT_FILES}"
+      else
+        echo "ERROR: There are no main extension files to remove from the base image"
+        echo "       It is possible that the extension that you inputted is not actually installed"
+        exit 1
+      fi
+      echo "----------------------------------UNINSTALLATION DONE----------------------------------"
+  done    
+fi
+
+# Compile gschema to include schemas from extensions  & to refresh schema state after uninstall is done
+echo "Compiling gschema to include extension schemas & to refresh the schema state"
 glib-compile-schemas "/usr/share/glib-2.0/schemas/" &>/dev/null
 
-echo "Finished the installation of Gnome extensions"
+echo "Finished the setup of gnome-extensions module"
```

Comment:
```suggestion
```

useless comment, CLI already sends a finish message, and its not the "setup of the module" that is finished but the module itself

Reply:
Yes, thanks